src/components/dock/dock.tsx

```diff
@@ -0,0 +1,186 @@
+ * @exampleComponent limel-example-dock-colors-css
+ * @private
+ */
+@Component({
+    tag: 'limel-dock',
+    shadow: true,
+    styleUrl: 'dock.scss',
+})
+export class Dock {
+    @Element()
+    public host: HTMLLimelDockElement;
+
+    /**
+     * Items that are placed in the dock.
+     */
+    @Prop()
+    public dockItems: DockItem[] = [];
+
+    /**
+     * Items that are placed at the bottom of the dock. (Or at the end in mobile
+     * layout.)
+     */
+    @Prop()
+    public footerItems: DockItem[] = [];
+
+    /**
+     * Defines the width of the component, when it loads.
+     * - `true`: shows both icons and labels of the Dock items.
+     * - `false`: only shows icons of the doc items, and displays
+     * their labels as tooltip.
+     *
+     * Note: when `useMobileLayout` is `true`, labels will always
+     * be shown as tooltips. Read more below…
+     */
+    @Prop({ reflect: true })
+    public expanded = false;
+
+    /**
+     * Set to `false` if you do not want to allow end-users
+     * to exapnd or shrink the Dock. This will hide the
+     * expand/shrink button, and the only things that defines
+     * the layout will be the `expanded` property, and
+     * the `mobileBreakPoint`.
+     */
+    @Prop({ reflect: true })
+    public allowResize = true;
+
+    /**
+     * Defines the breakpoint in pixles, at which the component will be rendered
+     * in a hoizontal layout. Default breakpoint is `700` pixels, which means
+     * when the screen size is smaller than `700px`, the component will automatically
+     * switch to a horizontal layout.
+     */
+    @Prop({ reflect: true })
+    public mobileBreakPoint: number = DEFAULT_MOBILE_BREAKPOINT;
+
+    /**
+     * A document may have several <nav> elements, for example,
+     * one for site navigation and one for intra-page navigation.
+     * `ariaLabelledBy` can be used in such case to promote accessibility,
+     */
+    @Prop({ reflect: true })
+    public ariaLabelledBy: string;
+
+    /**
+     * Fired when a Dock item has been selected from the dock.
+     */
+    @Event()
+    private selected: EventEmitter<DockItem>;
+
+    /**
+     * Is used to render the component horizontally, and place
+     * the Dock items in a row.
+     */
+    @State()
+    private useMobileLayout = false;
+
+    /**
+     * Used to trigger mobileLayout, when viewport width is changed.
+     */
+    private observer: ResizeObserver;
+
+    public componentDidLoad() {
+        this.observer = new ResizeObserver(this.handleResize);
+        this.observer.observe(document.body);
+    }
+
+    disconnectedCallback() {
+        this.observer.disconnect();
+    }
+
+    public render() {
+        return (
+            <Host
+                class={{
+                    dock: true,
+                    expanded: this.expanded,
+                    'has-mobile-layout': this.useMobileLayout,
+                }}
+            >
+                <nav aria-labelledby={this.ariaLabelledBy}>
```

Comment:
`aria-labelledby` is not the correct attribute to use:

> The [aria-labelledby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attribute identifies the element (or elements) that labels the element it is applied to.

We could use `aria-label`, or we could put the text into an element, give it an `id`, and then refer to that `id` with `aria-labelledby` 🙂

If it's more of a description than a label, we could use `aria-describedby`. `aria-description` also exists, but apparently isn't ready for live use yet.

> The [aria-description](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-description) attribute is similar to [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) in that both provide a text string to associate with the element, but a label should be short and concise, while the description can be longer as it is intended to provide more context and information.

> Note: [aria-description](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-description) is still in W3C Editor's Draft for ARIA 1.3. For the time being, continue to use [aria-describedby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby), which has been supported since ARIA 1.1.

Reply:
Aha ok.
They said:
> A document may have several [<nav>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav) elements, for example, one for site navigation and one for intra-page navigation. [aria-labelledby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) can be used in such case to promote accessibility, see [example](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#labeling_section_content).

But then I didn't check their example 🙈 👍 

```html
<header>
  <nav aria-labelledby="primary-navigation">
    <h2 id="primary-navigation">Primary navigation</h2>
    <!-- navigation items -->
  </nav>
</header>

<!-- page content -->

<footer>
  <nav aria-labelledby="footer-navigation">
    <h2 id="footer-navigation">Footer navigation</h2>
    <!-- navigation items -->
  </nav>
</footer>
```

Shall we go for `aria-label` instead?

Reply:
⚡

I've changed to `aria-label`, and also made the property required. Is it too much to make it required? We want to strongly encourage accessibility, but maybe a label isn't always necessary to make the functionality clear? Also, it may be better to have no label than have some really bad label that someone just added because the property was required…?

Hmm… I think I'm changing my mind on this required thing… 😅

Reply:
@Kiarokh Thoughts on this prop being required or not?

Reply:
I made it optional again.

Reply:
Well, I think we can't really play the Moral Police here. 😄 There are many things that are "required", when it comes to accessibility; but people skip them and we can't do much. 

For this specific case of `limel-dock`, the accessibility-damage of not providing a required aria-label is not so big. The element is already using `nav` which makes it way more accessible. Providing `aria-label` is good specially when there are multiple navigation methods. Since this Dock thingy is most probably used within an "app UI" (not a website UI), there is good reasons to believe that there is only this primary navigation way. (unless folks are using something like a Breadcrumbs elements or similar. 

Anyhow, I think we can easily leave the responsibility to the user here. Our responsibility was providing this prop.

Nevertheless, if we wanna play the Moral Police and provide something like this as "required" –which we think that people may not provide a proper value for– I'd say we can already be a good police and fallbacks in place (in translation files) ourselves.

We would need these anyways for our own CRM, so I already have collected them via [Microsoft Language portal](https://www.microsoft.com/en-us/language/Search?&searchTerm=primary%20navigation&langID=750&Source=true&productid=0).

<table>
<tr>
	<td>🇬🇧 en: 
	<td>Primary navigation
<tr>
	<td>🇸🇪 sv: 
	<td>Primär navigering
<tr>
	<td>🇳🇴 no: 
	<td>Primærnavigasjon
<tr>
	<td>🇩🇰 da: 
	<td>Primaire navigatie
<tr>
	<td>🇫🇮 fi: 
	<td>Ensisijainen siirtyminen
<tr>
	<td>🇩🇪 de: 
	<td>Primäre Navigation
<tr>
	<td>🇫🇷 fr: 
	<td>Navigation principale
<tr>
	<td>🇳🇱 nl: 
	<td>Primaire navigatie
</table>